tests/schema/test_types.py

```diff
@@ -57,3 +57,47 @@ def test_type_relationships():
     assert len(myobj.children_as_dict) == 2
     assert myobj.children_as_dict['child1'].parent == myobj.children_as_dict['child2'].parent
     assert myobj.children_as_dict['child1'].parent.parent == myobj
+
+def test_case_insensitive_string():
+    # Core functionality we are trying to achieve
+    assert String("Steve") == String("steve")
+    assert String("Steve") != String("burns")
+
+    # Comparison to built-in str works 
+    # both ways as well !!!
+    assert "sTeVe" == String("Steve")
+    assert String("Steve") == "sTeVe"
+    assert "burns" != String("Steve")
+    assert String("Steve") != "burns"
+
+    # As long as parent string is of type
+    # `String`, contains works as expected
+    assert String("Ste") in String("steve")
+    assert "sTeVe" in String("steve")
+    assert "StE" in String("steve")
+    assert "Burns" not in String("steve")
+    # WARNING: parent string MUST be of type
+    # `String` for contain to work. See below
+    assert String("ste") not in "Steve"
+
+    # WARNING: Object behaves like built-in str
+    # for almost everything else
+
+    # Slicing returns built-in str (not `String`)
+    x, y = String("Steve")[0], String("Steve")[0:3]
+    assert type(x) == str
+    assert type(y) == str
+    assert x == 'S'
+    assert y == 'Ste'
+    # Reverse via slicing works returns
+    # built-in str (not `String`)
+    x = String("Steve")[::-1]
+    assert type(x) == str
+    assert x == "evetS"
```

Comment:
Can use a second pair of eyes here @stevenmburns. It is possible to override this behavior to ensure slicing a SpiceStr returns object of type SpiceStr as well. Should we? (I still have to refactor `String` to `SpiceStr` in this PR by the way).

Reply:
My take is that I wouldn't worry about this until we see it causing a problem. The first few times we doit (if we every do), we change explicitly convert it to a SpiceStr. If we do it often, then we can add the conversion, can clean up the extra code when we did the conversion explicitly. I suspect we won't do this often.